File: 1777-determine-if-two-strings-are-close/determine-if-two-strings-are-close.go
func closeStrings(word1 string, word2 string) bool {
    freq1 := [26]int{}
    freq2 := [26]int{}

    for _, c := range word1 {
        freq1[c - 'a']++
    }

    for _, c := range word2 {
        freq2[c - 'a']++
    }

    for i := 0; i < 26; i++ {
        if (freq1[i] > 0) != (freq2[i] > 0) {
            return false
        }
    }

    sort.Ints(freq1[:])
    sort.Ints(freq2[:])

    for i := 0; i < 26; i++ {
        if freq1[i] != freq2[i] {
            return false;
        } 
    }
    return true;
}

File: 1777-determine-if-two-strings-are-close/determine-if-two-strings-are-close.cpp
class Solution {
public:
    bool closeStrings(string word1, string word2) {
        vector<int> freq1(27, 0), freq2(27, 0);

        for(auto c : word1) freq1[c - 'a']++;
        for(auto c : word2) freq2[c - 'a']++;

        for (int i = 0; i < 26; i++) {
            if (freq1[i] > 0 ^ freq2[i] > 0) {
                return false;
            }
        }

        sort(freq1.begin(), freq1.end());
        sort(freq2.begin(), freq2.end());
        
        for(int i = 0; i < 27; ++i) {
            if(freq1[i] != freq2[i]) return false;
        }

        return true;
    }
};